22.md

```diff
@@ -0,0 +1,184 @@
+NIP-22
+======
+
+Comment
+-------
+
+`draft` `optional`
+
+A comment is a threading note always scoped to a root event or an `I`-tag.
+
+It uses `kind:1111` with plaintext `.content` (no HTML, Markdown, or other formatting).
+
+Comments MUST point to the root scope using uppercase tag names (e.g. `K`, `E`, `A` or `I`)
```

Comment:
I think it should be a SHOULD instead of a MUST. Clients don't always have access to the root event.

Reply:
This only works when commenting on events that use this spec and you would have to trust the parent author to set the correct root data. 

A comment only needs its parent data. Adding root data might be helpful for clients implementing full threads but is burdensome for clients that just want to create a comment. It should be encouraged but not be a requirement.

Reply:
The root data is not even that useful. Right now you would get the whole thread by querying `E` because most threads are tiny. When threads get bigger you would hit relay restrictions and only receive a subset of the comments many of which will be n-level instead top-level replies, which are not useful for building the thread from the root.

Reply:
What should be set when the parent does not set an `E` tag (we shouldn't rely on other people complying) and you also don't know the root? 
Having the root scope as a MUST requirement doesn't make sense in this case. It should be SHOULD.

I personally don't really care about it as I'm going to disrespect the root scope the same way I disrespect nip10, but when a library dev sees a MUST he will implement it that way and I would like to use a library function instead of building the event myself.